libs/core/lci_base/src/lci_environment.cpp

```diff
@@ -214,17 +214,22 @@ namespace hpx { namespace util {
             LCI_initialized(&lci_init);
             if (lci_init)
             {
-                HPX_ASSERT(prg_thread_flag.load() == true);
-                HPX_ASSERT(prg_thread_p != nullptr);
-                prg_thread_flag = false;
-                prg_thread_p->join();
-                delete prg_thread_p;
-
+                join_prg_thread_if_running();
                 LCI_finalize();
             }
         }
     }
 
+    void lci_environment::join_prg_thread_if_running()
+    {
+        if (prg_thread_flag.load())
+        {
+            HPX_ASSERT(prg_thread_p != nullptr);
+            prg_thread_flag = false;
+            prg_thread_p->join();
```

Comment:
Can the thread (-object) be deleted after joining?

Reply:
Yes, will do.

Reply:
done.